### ISSUE TITLE
Policies are accessible to all connectors

### ISSUE THREAD

**Issue by trueg** [FraunhoferISST/DataspaceConnector#478]
While not necessarily a bug but rather a design issue I am posting it is a bug anyway to get the discussion started somewhere:

When negotiating a new contract a connector can list the policies of any resource in full, including all details. In fact, these details are even required as API input. 

IMHO this poses a serious security risk as anyone can essentially figure out who has access to what data. I doubt that this is acceptable to the legal departments of any bigger company.

Thus, I propose to simply hide the permission details from consumer connectors. They do not need to know why they have access or not. Instead of requiring the full permission rule as API input just stick to the requested access type (like idsc:use).

Side note: Why is it possible to negotiate a contract for an artifact I do not have access to anyway?

**Comment by gregors101** [comment]
I agree with the point that asking about permission details is a security leak. Any consumer is able to ask any provider about permission details, use the permission details for creating contracts and get the data from the provider (assumed both are valid connectors with valid certificates and the endpoints are known).

A possible solution could be, that the DAPS authorize the communication between specific connectors and therefore asking about the permission information is allowed?

**Comment by trueg** [comment]
After a lot of thinking I have to change my opinion on this: the only issue are in fact usage permissions in the form of white lists, etc.. These, however, should not really be part of the contract offer to begin with. I would assume that the idea is rather that contract offers come with a set of conditions (just like contracts "in the real world") ranging from "consumer needs an account at xyz" to "by agreeing to this contract the consumer consents to a monthly fee of N". The current restriction of contract offers to specific consumer connectors is, thus, only a stop-gap until we have support for these more sophisticated policies.

That said, I still think that, given the lack of policy support atm, it would be good to have a separate set of permissions for a contract offer which are in fact private (not visible to consumer connectors) and state which connectors are allowed to negotiate a contract based on an offer.

This might require an extension to the IDS model, that I am not sure about atm.

In summary:
- contract offers should not be private, my original claim was simply wrong
- contract offers should have a separate, private set of permissions which the provider connector can use to determine which consumer is allowed to negotiate a contract (stop-gap until we have full policy support with payment models, etc.)

**Comment by trueg** [comment]
Closing this issue for good now as I realized that contracts are actually only useful with proper policies that do not restrict the connector via a white list etc..
